OpenRA.Mods.Common/Traits/Attack/AttackFrontal.cs

```diff
@@ -17,7 +17,8 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Unit got to face the target")]
 	public class AttackFrontalInfo : AttackBaseInfo, Requires<IFacingInfo>
 	{
-		public readonly int FacingTolerance = 0;
+		[Desc("Tolerance for attack angle. Range [0, 128], 128 covers 360 degrees.")]
+		public readonly new int FacingTolerance = 0;
```

Comment:
I've always considered the `new` modifier to be a code smell, so we tend to avoid it as a style convention.  If we remove the `FacingTolerance` definition on `AttackBaseInfo` there is only one place in this PR that can't trivially change to use the subclass values directly - the `TargetInFiringArc` check in `AutoTarget`.

If i'm not mistaken, this check is never actually used (`AllowTurning` is always `True` in the default mods, and units in `HoldFire` shouldn't be choosing targets anyway), so its not clear why this is needed, and why we can't stick with the subclasses defining `FacingTolerance` themselves.

Reply:
This check is used by `AttackAircraft` via `AttackFollow` to select only opportunity targets within it's firing arc, so it shouldn't be removed.

Letting the subclasses override the `FacingTolerance` parameter is obviously a hack, but the way forward is to define it only in the baseclass. Removing it from the baseclass is backwards. The whole point was to implement a hacky stopgap so that the proper Attack* refactor can be done in it's own PR and to not delay other PRs that depend on this. I do not want to undo changes that I would then have to redo again and I do not intend for this hack to be here when the next release comes.